### PR TITLE
Update ch3

### PR DESCRIPTION
**Pull request by danielvdende**

### CODE REVIEW

File: chapter03/dags/01_unscheduled.py [hunk 1]
```diff
@@ -4,41 +4,42 @@
...
     fetch_events = BashOperator(
         task_id="fetch_events",
         bash_command=(
-            "curl -o /data/events.json "
-            "http://events_api:5000/events?"
-            "start_date=2024-01-01&"
-            "end_date=2024-01-02"
+            "mkdir -p /data/01_unscheduled && "
+            "curl -o /data/01_unscheduled/events.json"
+            " http://events-api:8081/events/latest"
```
Comment (krisgeus) [inline chapter03/dags/01_unscheduled.py:33]: Compose says port 5000?

File: chapter03/dags/01_unscheduled.py [hunk 2]
```diff
@@ -4,41 +4,42 @@
...
 ):
     fetch_events = BashOperator(
         task_id="fetch_events",
         bash_command=(
-            "curl -o /data/events.json "
-            "http://events_api:5000/events?"
-            "start_date=2024-01-01&"
-            "end_date=2024-01-02"
+            "mkdir -p /data/01_unscheduled && "
+            "curl -o /data/01_unscheduled/events.json"
```
Comment (krisgeus) [inline chapter03/dags/01_unscheduled.py:32]: spaces in different places. First line EOL, second line Start of line, third line EOL again

File: chapter03/dags/02_daily_schedule.py [hunk 3]
```diff
@@ -4,41 +4,45 @@
...
+    end_date=pendulum.datetime(year=2024, month=1, day=5),
 ):
     fetch_events = BashOperator(
         task_id="fetch_events",
         bash_command=(
-            "curl -o /data/events.json "
-            "http://events_api:5000/events?"
-            "start_date={{data_interval_start | ds}}&"
-            "end_date={{data_interval_end | ds}}"
+            "mkdir -p /data/02_daily_schedule && "
```
Comment (krisgeus) [inline chapter03/dags/02_daily_schedule.py:34]: Same spaces comment here

File: chapter03/dags/02_daily_schedule.py [hunk 4]
```diff
@@ -4,41 +4,45 @@
...
+            "curl -o /data/02_daily_schedule/events.json"
+            " http://events-api:8081/events/latest"
```
Comment (krisgeus) [inline chapter03/dags/02_daily_schedule.py:36]: portnumber 5000

File: chapter03/dags/03_timedelta.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+with DAG(
+    dag_id="03_timedelta",
+    schedule=datetime.timedelta(days=2),
+    start_date=pendulum.datetime(year=2024, month=1, day=1),
+    end_date=pendulum.datetime(year=2024, month=1, day=5),
+):
+    fetch_events = BashOperator(
+        task_id="fetch_events",
+        bash_command=(
+            "mkdir -p /data/03_timedelta && "
```
Comment (krisgeus) [inline chapter03/dags/03_timedelta.py:34]: Spaces and portnumber also here

File: chapter03/dags/01_unscheduled.py (hunk 1 above)
Comment (danielvdende) [reply chapter03/dags/01_unscheduled.py]: Nice catch! The API container itself actually exposes 8081, so you end up with both 5000 and 8081:
```
0.0.0.0:5000->5000/tcp, 8081/tcp
```
All code was using 8081, so I've switched it over in the compose now too.

File: chapter03/dags/01_unscheduled.py (hunk 2 above)
Comment (danielvdende) [reply chapter03/dags/01_unscheduled.py]: Fixed.

File: chapter03/dags/03_timedelta.py (hunk 5 above)
Comment (danielvdende) [reply chapter03/dags/03_timedelta.py:34]: Fixed.

**Review by krisgeus — APPROVED** [review]

**Pull request merged**